contract/src/airdrop/prepareAirdropKit.js

```diff
@@ -0,0 +1,341 @@
+} from '@agoric/ertp';
+import {
+  atomicRearrange,
+  fromOnly,
+  provideAll,
+  toOnly,
+} from '@agoric/zoe/src/contractSupport/index.js';
+import { E, Far } from '@endo/far';
+import { makePromiseKit } from '@endo/promise-kit';
+import { AIRDROP_ADMIN_MESSAGES, CLAIM_MESSAGES } from './helpers/messages.js';
+import '../../types.js';
+import { makeCancelTokenMaker } from './helpers/time.js';
+
+const head = ([x, ...xs]) => x;
+
+const makeTracer = label => value => {
+  console.log(label.toUpperCase(), '::::', value);
+  return value;
+};
+
+const tracer = makeTracer('Airdrop airdropCampaign');
+/** @type {airdropCampaignMeta} */
+export const meta = {
+  upgradability: 'canUpgrade',
+  terms: {
+    proofHolderFacet: M.remotable('proofHolder Powers'),
+  },
+  privateArgsShape: M.splitRecord({
+    powers: {
+      timerService: M.eref(M.remotable('TimerService')),
+    },
+  }),
+};
+
+harden(meta);
+
+const handleFirstIncarnation = (baggage, key) =>
+  !baggage.has(key)
+    ? baggage.init(key, 1)
+    : baggage.set(key, baggage.get(key) + 1);
+
+const AirdropIssuerDetailsShape = harden({
+  brand: BrandShape,
+  issuer: IssuerShape,
+});
+
+const finalMetrics = (promise, purse) =>
+  promise.resolve({
+    remainingTokens: purse.getCurrentAmount(),
+  });
+const startupAssertion = (arg, keyName) =>
+  assert(
+    arg,
+    `Contract has been started without required property: ${keyName}.`,
+  );
+
+const makeWaker = (name, func) => {
+  return Far(name, {
+    wake: timestamp => func(timestamp),
+  });
+};
+
+const createWakeup = async (timer, wakeUpTime, timeWaker, cancelTokenMaker) => {
+  const cancelToken = cancelTokenMaker();
+  await E(timer).setWakeup(wakeUpTime, timeWaker, cancelToken);
+};
+
+export const start = async (zcf, privateArgs, baggage) => {
+  // assertIssuerKeywords(zcf, harden(['Airdroplets']));
+
+  const { issuers, rootHash: merkleRoot, claimWindowLength } = zcf.getTerms();
+
+  const { Airdroplets } = issuers;
+
+  console.log('PEELING OFF FROM TERMS.ISSUERs', { Airdroplets });
+  // TODO handle fail cases?
+  tracer('inside prepareAirdropKit');
+
+  console.log(zcf.getTerms(), { privateArgs });
+  handleFirstIncarnation(baggage, 'airdropCampaignVersion');
+  startupAssertion(
+    privateArgs.distributionSchedule,
+    'privateArgs.distributionSchedule',
+  );
+  startupAssertion(
+    privateArgs.claimWindowStartTime,
+    'privateArgs.claimWindowStartTime',
+  );
+  startupAssertion(privateArgs.purse, 'privateArgs.purse');
+  startupAssertion(privateArgs.timer, 'privateArgs.purse');
+
+  startupAssertion(merkleRoot, 'terms.rootHash');
+  startupAssertion(claimWindowLength, 'terms.claimWindowLength');
+  tracer('privateArgs:'.concat(privateArgs.toString()));
+
+  const terms = zcf.getTerms();
+
+  const { zcfSeat: contractSeat } = zcf.makeEmptySeatKit();
+
+  console.log('TERMS', terms);
+
+  const { distributionSchedule, timer } = privateArgs;
+
+  const {
+    issuers: { Airdroplets: AirdropIssuer },
```

Comment:
now both `Airdroplets` and `AirdropIssuer` are bound to this issuer... I wonder why 2 variables for the same thing.

Reply:
This has since been removed.